cli/responseemitter.go

```diff
@@ -168,29 +210,15 @@ func (re *responseEmitter) Emit(v interface{}) error {
 		return io.ErrClosedPipe
 	}
 
-	if err, ok := v.(cmdkit.Error); ok {
-		v = &err
-	}
-
 	var err error
 
 	switch t := v.(type) {
 	case *cmdkit.Error:
-		_, err = fmt.Fprintln(re.stderr, "Error:", t.Message)
-		if t.Code == cmdkit.ErrFatal {
-			defer re.Close()
-		}
-		re.wLock.Lock()
-		defer re.wLock.Unlock()
-		re.exit = 1
-
-		return nil
-
+		log.Errorf("got an error: %v\n%s", err, debug.Stack())
 	case io.Reader:
 		_, err = io.Copy(re.stdout, t)
 		if err != nil {
-			re.SetError(err, cmdkit.ErrNormal)
-			err = nil
+			return err
```

Comment:
I'm wondering if this should also call `CloseWithError`. We could also say that it's the callers responsibility to return the error from `Run`. Thoughts?

Reply:
Calling `CloseWithError` here would close the emitter and only print `io.Copy`'s error. I think it's better to allow the caller to specify a more intelligible error.

Reply:
Fair enough.